app/src/main/java/com/github/se/gomeet/ui/authscreens/register/RegisterUsernameEmail.kt

```diff
@@ -0,0 +1,156 @@
+import androidx.compose.material3.Text
+import androidx.compose.material3.TextField
+import androidx.compose.material3.TextFieldColors
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.rememberCoroutineScope
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.platform.LocalConfiguration
+import androidx.compose.ui.text.input.ImeAction
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.unit.dp
+import com.github.se.gomeet.model.user.GoMeetUser
+import com.github.se.gomeet.ui.theme.DarkGrey
+import com.github.se.gomeet.viewmodel.UserViewModel
+import kotlinx.coroutines.launch
+
+/**
+ * This composable function collects username and email from the user. It performs validations on
+ * the input and advances the registration process upon successful validation.
+ *
+ * @param callback Function to be called with the username and email when both inputs are valid and
+ *   not already taken.
+ * @param userViewModel ViewModel to access and interact with user data.
+ * @param textFieldColors Custom colors for the TextField components used in this Composable.
+ */
+@Composable
+fun RegisterUsernameEmail(
+    callback: (String, String) -> Unit,
+    userViewModel: UserViewModel,
+    textFieldColors: TextFieldColors
+) {
+  val screenHeight = LocalConfiguration.current.screenHeightDp.dp
+  var username by remember { mutableStateOf("") }
+  var email by remember { mutableStateOf("") }
+  var isValidEmail by remember { mutableStateOf(false) }
+  var isValidUsername by remember { mutableStateOf(false) }
+  var firstClick by remember { mutableStateOf(true) }
+  var charactersExceeded by remember { mutableStateOf(false) }
+  var allUsers by remember { mutableStateOf<List<GoMeetUser>?>(null) }
+  val coroutineScope = rememberCoroutineScope()
+
+  LaunchedEffect(Unit) { coroutineScope.launch { allUsers = userViewModel.getAllUser() } }
+
+  Column(
+      modifier = Modifier.fillMaxSize(),
+      horizontalAlignment = Alignment.CenterHorizontally,
+      verticalArrangement = Arrangement.SpaceAround) {
+        Text(
+            text = "Welcome to GoMeet !\nPlease enter a username and an email.",
+            modifier = Modifier.fillMaxWidth(),
+            style = MaterialTheme.typography.titleMedium,
+            textAlign = TextAlign.Center)
+
+        Spacer(modifier = Modifier.size(screenHeight / 20))
+
+        TextField(
+            value = username,
+            onValueChange = {
+              if (it.length < 26) {
+                charactersExceeded = false
+                username = it
+              } else {
+                charactersExceeded = true
+                username = it
```

Comment:
Setting username here even if it's too long (even if you can't progress to the next screen) might introduce bugs, better to just set charactersExceeded to true in the else condition

Reply:
In this case, the user will never exceed the maximum number of characters for a username (he will not be able to write more than 26 characters) and the `charactersExceeded` variable will be useless. I think we can leave like that to stay consistent with how other inputs are handled  (the user can write whatever he wants in the field, but the registration process is blocked if the input is not valid).